pkg/dockerregistry/server/auth.go

```diff
@@ -535,3 +545,11 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func isMetricsBearerToken(ctx context.Context, token string) bool {
+	config := ConfigurationFrom(ctx)
+	if config.Metrics.Enabled {
+		return config.Metrics.Secret == token
+	}
```

Comment:
@smarterclayton In the beginning, we discussed this already and came to the decision that we would use the shared secret for metrics endpoint because router doing it in same way.

https://github.com/openshift/origin/pull/12711#issuecomment-276216300
https://github.com/openshift/origin/pull/12711#issuecomment-276679344